pkg/controller/node/status.go

```diff
@@ -281,16 +296,26 @@ func calculateStatus(fg featuregates.FeatureGate, mcs []*mcfgalphav1.MachineConf
 	return status
 }
 
-func getPoolUpdateLine(pool *mcfgv1.MachineConfigPool) string {
+func isPinnedImageSetNodeUpdating(mcs *mcfgalphav1.MachineConfigNode) bool {
+	var updating int32
+	for _, set := range mcs.Status.PinnedImageSets {
+		if set.CurrentGeneration != set.DesiredGeneration {
+			updating++
```

Comment:
suggestion (non-blocking): Once we encounter the first condition where `set.CurrentGeneration != set.DesiredGeneration`, we could probably return `true` early since we don't need to examine the rest of the conditions. Only once we've examined all of the conditions and found none to be unequal, we can return `false`.

(Does not need to be addressed right now.)